### PR TITLE
Update availability endpoint in 11ty to match registrar

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Changes proposed in this pull request:

References [this PR](https://github.com/cisagov/manage.get.gov/pull/1439) for updating the availability endpoint in the registrar + updated error messaging for if a user adds `.` in their to be domain. Please note in the above PR, the error message of `Enter the .gov domain you want without any periods.` will display when you click "Save and Continue" which is correct -- currently the error messages are not synced and we are aware of that.

image

This work will be done in a diff PR:
This PR is for updating the availability endpoint on the beta site + some error messaging changes (see [this Slack discussion](https://cisa-corp.slack.com/archives/C05BDEA3C11/p1701824406852919) for adding in the error messaging).

## security considerations
[Note the any security considerations here, or make note of why there are none]

### CODE REVIEW

File: _includes/domain-search.html [hunk 1]
```diff
@@ -27,7 +27,7 @@
                 <div class="usa-alert usa-alert--success usa-alert--slim" role="alert">
                     <div class="usa-alert__body">
                         <p class="usa-alert__text">
-                        <b>${requested_domain_display_string}</b> is available!
+                        <b>${requested_domain_display_string}</b> is available !
```
Comment (erinysong) [inline _includes/domain-search.html:30]: nonblocking: Is this space before the ! supposed to be there?

Comment (therealslimhsiehdy) [reply _includes/domain-search.html]: Great catch -- fixed in [this commit](https://github.com/cisagov/getgov-home/pull/136/commits/1c09bdcd73032c187b6de0a89f44e131dcc10ffb)!

File: _includes/domain-search.html [hunk 2]
```diff
@@ -43,9 +43,10 @@
                     <div class="usa-alert__body">
                         <p class="usa-alert__text">
                             <b>${requested_domain_display_string}</b> isn't available.
-                            <a class="usa-link" href="{{ '/domains/choosing/' | url }}">
-                                Read more about choosing your .gov domain.
-                            </a>
+                            ${response.message} 
+                        <a class="usa-link" href="{{ '/domains/choosing/' | url }}">
```
Comment (erinysong) [inline _includes/domain-search.html:47]: also a small formatting nit: I think the <a> tags should be indented to be aligned with line 46!

**Comment by therealslimhsiehdy** [comment]
> The changes here LGTM! One nonblocking edge-case question is how we want to handle periods for domains - for example, is there a reason for subdomains (city.meowardcounty.gov) to ever be inputted in this form and if there is a case we'd want to check subdomain availability here, how would we want to handle that?

FWIW that's a great call out! Are we only approving top level domains currently, and not checking for subdomains right now? @dave-kennedy-ecs tagging you in case you know as it seems like with the new error changes you merged earlier today!

**Comment by h-m-f-t** [comment]
>  is there a reason for subdomains (city.meowardcounty.gov) to ever be inputted in this form

Individual domain registrants manage their own subdomains, and they are usually tightly controlled by the org. Users _might_ enter subdomains here, but we should simply warn and say something like "only second-level domains (example.gov) can be requested".

**Review by dave-kennedy-ecs — APPROVED** [review]

**Pull request merged**